Doc/library/ssl.rst

```diff
@@ -1730,6 +1778,14 @@ to speed up repeated connections from the same clients.
    The protocol version chosen when constructing the context.  This attribute
    is read-only.
 
+.. attribute:: SSLContext.host_flags
```

Comment:
Does this need to be a public API? I feel like we can set reasonable defaults that are applicable to everything.

Reply:
I exposed the flag primary for https://github.com/shazow/urllib3/issues/497. urllib3 wants to disable CN support and require certificates with SAN fields.

Reply:
What if we just expose a single `check_subject_hostname` or something like that, instead of all these flags?

Reply:
I'm ok with that, but I'm not 100% happy with the name.

* ``check_subject_cn``
* ``check_common_name``
* ``check_subject_common_name``
* ``use_common_name``
* ``check_only_san``
* ...

I'm going to rename ``host_flags`` to ``_host_flags``, keep all flags in ``_ssl`` and implement the flag in pure Python. That way we don't expose the API, but still have it available to implement other flags.

Reply:
Sounds good -- my desire is for us not to commit to a really broad API when we really want a tiny thing.

`hostname_checks_common_name` perhaps?

Reply:
That's explicit, almost German. I like it :)

The feature is only available with OpenSSL 1.1.0+. How should I handle OpenSSL 1.0.2? Don't define the property at all or only implement readonly property + ``ssl.HAS_NEVER_CHECK_HOSTNAME``?

Reply:
readonly + a constant makes sense to me

Reply:
Done

I'm currently at a conference and will take care of proper documentation next week.